trieste/models/config.py

```diff
@@ -122,50 +165,65 @@ class ModelConfig:
     model: Any
     """
     The low-level model to pass to the :class:`~trieste.models.TrainableProbabilisticModel`
-    interface registered with the `model` via :class:`ModelRegistry`. The model has to be one of
+    wrapper registered with the `model` via :class:`ModelRegistry`. The model has to be one of
     the supported models, that is, registered via :class:`ModelRegistry`. We use type `Any` here as
     this can be either a model that is supported by default (for example, GPflow- or GPflux-based
     models) or a user-defined model that has been registered.
     """
 
     model_args: dict[str, Any] = field(default_factory=lambda: {})
     """
-    The keyword arguments to pass to the model interface
+    The keyword arguments to pass to the model wrapper
     :class:`~trieste.models.TrainableProbabilisticModel` registered with the `model` via
     :class:`ModelRegistry`.
     """
 
-    optimizer: Any = field(default_factory=lambda: tf.optimizers.Adam())
+    optimizer: Any = None
     """
-    The low-level optimizer to pass to the :class:`~trieste.models.Optimizer` interface
+    The low-level optimizer to pass to the :class:`~trieste.models.Optimizer` wrapper
     registered with the `model` via :class:`ModelRegistry`, with which to train the model (by
-    minimizing its loss function).
+    minimizing its loss function). The model has to be one of the supported models, that is,
+    registered via :class:`ModelRegistry`. We use type `Any` here as this can be either an
+    optimizer that is supported by default (for example, GPflow or TensorFlow) or a user-defined
+    optimizer that has been registered.
     """
 
     optimizer_args: dict[str, Any] = field(default_factory=lambda: {})
     """
-    The keyword arguments to pass to the optimizer interface :class:`~trieste.models.Optimizer`
+    The keyword arguments to pass to the optimizer wrapper :class:`~trieste.models.Optimizer`
     registered with the `model` via :class:`ModelRegistry`.
     """
 
     def __post_init__(self) -> None:
         self._check_model_type()
+        self._check_optimizer_type()
 
     def _check_model_type(self) -> None:
         if type(self.model) in ModelRegistry.get_registered_models():
             return
-
         raise NotImplementedError(f"Not supported type {type(self.model)}")
 
+    def _check_optimizer_type(self) -> None:
+        if self.optimizer is not None:
+            for optimizer_type in type(self.optimizer).__mro__:
```

Comment:
Isn't it enough to call `ModelRegistry.get_optimizer_wrapper(optimizer_type)` here? (If you're concerned about the exception type, you could always catch and rethrow. Would be nice to keep the `__mro__` references to a minimum.)

Reply:
you are right, thanks!